report/src/trends/index.js

```diff
@@ -17,40 +17,40 @@ async function generateTrends(gitPath, latestReport, maxDaysHistory = 30) {
 
   // get history for the report file
   core.info(`Open GIT history for report.json`);
-  /** @type {GitHistory} */
   const history = await gitHistory(`${gitPath}/report.json`);
 
   // ensure git history is sorted from latest to oldest commit
-  history.sort((a, b) => b.commit.date() - a.commit.date());
+  history.sort((a, b) => b.commit.date().valueOf() - a.commit.date().valueOf());
 
   // will only extract N days of history
   const startDate = new Date(
     new Date().getTime() - maxDaysHistory * 24 * 60 * 60 * 1000
   );
 
+  /** @type {import("nodegit").Revwalk.HistoryEntry[]} */
+  const reduceInit = [];
   // compile summary contents from reports
   const commits = await Promise.all(
     history
       // include only relevant commits (in the date range)
       .reduce(
-        /** @param {GitHistoryEntry[]} filteredCommits*/ (
+       (
           filteredCommits,
           entry,
-          i,
-          all
+          i
         ) => {
           const isAfter = entry.commit.date() >= startDate;
           const isFirstBefore = !isAfter && i === filteredCommits.length;
-          // include relvant commits
+          // include relevant commits
           if (isAfter || isFirstBefore) {
             filteredCommits.push(entry);
           }
           return filteredCommits;
         },
-        []
+        reduceInit
       )
       // extract summary content for each commit
-      .map(async ({ /** @type {Commit} */ commit }) => {
+      .map(async ({ commit }) => {
         commit.repo = repo; // for some reason this is not populated by default and prevents `getEntry`
```

Comment:
this seems off, `Commit` does not have a property `repo`
https://www.nodegit.org/api/#commit

This may not be needed in newer versions of NodeGit.
If it is requited, seems like it may be a bug upstream in NodeGit.

Reply:
Yes, theres a weird bug or misusage, will report o NodeGit

Reply:
With `commit.repo` it doesnt work in this example : 

```js

const test = async (gitPath) => {
    const history = await gitHistory(`${gitPath}/report.json`);
    history.map(async ({ commit }) => {
        const treeEntry = await commit.getEntry("report.json");
        const blob = await treeEntry.getBlob();
    })
}

test(process.env.GIT_PATH);
```

it fails with

```
/xxx/dashlord-actions/report/node_modules/nodegit/dist/commit.js:290
  return this.repo.getTree(this.treeId());
                   ^

TypeError: Cannot read property 'getTree' of undefined
    at Commit.getTree (/xxx/dashlord-actions/report/node_modules/nodegit/dist/commit.js:290:20)
    at Commit.getEntry (/xxx/dashlord-actions/report/node_modules/nodegit/dist/commit.js:236:15)
```